include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -830,11 +829,34 @@ class LAPACKFullMatrix : public TransposeTable<number>
   compute_inverse_svd_with_kernel(const unsigned int kernel_size);
 
   /**
-   * Retrieve eigenvalue after compute_eigenvalues() was called.
+   * Retrieve eigenvalue after compute_eigenvalues() was called. The return
+   * type is always a complex number: In case the underlying matrix is
+   * real-valued, the type `std::complex<number>` is returned, whereas
+   * `number` is returned for complex numbers.
    */
-  std::complex<number>
+  std::complex<typename numbers::NumberTraits<number>::real_type>
   eigenvalue(const size_type i) const;
```

Comment:
This is one of the mentioned interfaces that were broken before.

Reply:
Nice catch!